lib/topic_query.rb

```diff
@@ -439,7 +439,11 @@ def create_list(filter, options = {}, topics = nil)
     topics = topics.to_a
 
     # These don't need to be shown in the topic list
-    topics = topics.reject { |t| [SiteSetting.privacy_topic_id, SiteSetting.tos_topic_id].include?(t[:id]) }
+    if (topics.pluck(:id) & [SiteSetting.privacy_topic_id, SiteSetting.tos_topic_id, SiteSetting.welcome_topic_id]).any?
```

Comment:
```suggestion
    if (topics.map(:id) & [SiteSetting.privacy_topic_id, SiteSetting.tos_topic_id, SiteSetting.welcome_topic_id]).any?
```

`topics` has already been loaded in memory at this point so we can avoid a DB query by not plucking.